src/Shared/Microsoft.AspNetCore.Razor.Test.Common/Language/Legacy/SyntaxNodeWriter.cs

```diff
@@ -94,10 +88,17 @@ private void WriteNode(SyntaxNode node)
 
         WriteChunkGenerator(node);
 
-        var annotation = node.GetAnnotations().FirstOrDefault(a => a.Kind == SyntaxConstants.EditHandlerKind);
-        if (annotation != null && annotation.Data is SpanEditHandler handler)
+        if (validateSpanEditHandlers)
+        {
+            var annotation = node.GetAnnotations().FirstOrDefault(a => a.Kind == SyntaxConstants.EditHandlerKind);
+            if (annotation != null && annotation.Data is SpanEditHandler handler)
+            {
+                WriteEditHandler(handler);
+            }
+        }
+        else
         {
-            WriteEditHandler(handler);
+            // Assert.All(node.GetAnnotations(), a => Assert.IsNotType<SpanEditHandler>(a.Data));
```

Comment:
?

Reply:
Can't enable this assert until we actually stop producing edit handlers. Should be later this week.